File: php/test/RCONTests.php
<?php
/**
 * This code is free software; you can redistribute it and/or modify it under
 * the terms of the new BSD License.
 *
 * Copyright (c) 2008-2011, Sebastian Staudt
 *
 * @license http://www.opensource.org/licenses/bsd-license.php New BSD License
 */

error_reporting(E_ALL & ~E_USER_NOTICE);

require_once dirname(__FILE__) . '/../lib/steam-condenser.php';

require_once 'PHPUnit/Framework.php';

/**
 * @author     Sebastian Staudt
 * @package    steam-condenser
 * @subpackage tests
 */
class RCONTests extends PHPUnit_Framework_TestCase {

    public function testRconLongGoldSrcServer() {
        $server = new GoldSrcServer(new InetAddress('127.0.0.1'));
        $server->rconAuth('test');
        $rconReply = $server->rconExec('cvarlist');
        echo "$rconReply\n";
        $this->assertTrue(
            strpos($rconReply, 'CvarList ? for syntax') !== false,
            'Did not receive complete cvarlist.');
    }

    public function testRconLongSourceServer() {
        $server = new SourceServer(new InetAddress('127.0.0.1'));
        $server->rconAuth('test');
        $rconReply = $server->rconExec('cvarlist');
        echo "$rconReply\n";
        $this->assertTrue(
            strpos($rconReply, 'total convars/concommands') !== false,
            'Did not receive complete cvarlist.');
    }

    public function testRconShortGoldSrcServer() {
        $server = new GoldSrcServer(new InetAddress('127.0.0.1'));
        $server->rconAuth('test');
        $rconReply = $server->rconExec('version');
        echo "$rconReply\n";
        $this->assertTrue(
            strpos($rconReply, 'Protocol version') !== false &&
            strpos($rconReply, 'Exe version') !== false &&
            strpos($rconReply, 'Exe build') !== false,
            'Did not receive correct version response.');
    }

    public function testRconShortSourceServer() {
        $server = new SourceServer(new InetAddress('127.0.0.1'));
        $server->rconAuth('test');
        $rconReply = $server->rconExec('version');
        echo "$rconReply\n";
        $this->assertTrue(
            strpos($rconReply, 'Protocol version') !== false &&
            strpos($rconReply, 'Exe version') !== false &&
            strpos($rconReply, 'Exe build') !== false,
            'Did not receive correct version response.');
    }
}

File: ruby/test/rcon_tests.rb
# This code is free software; you can redistribute it and/or modify it under the
# terms of the new BSD License.
#
# Copyright (c) 2008, Sebastian Staudt
#
# $Id$

$:.push File.join(File.dirname(__FILE__), "../lib")

require "ipaddr"
require "steam/servers/goldsrc_server"
require "steam/servers/source_server"
require "test/unit"

class RCONTests < Test::Unit::TestCase
  
  # This test tries to run the "status" command over RCON on a GoldSrc server
  def test_long_rcon_goldsrc_server
    assert_nothing_raised do
      server = GoldSrcServer.new IPAddr.new("127.0.0.1")
      server.rcon_auth "test"
      rcon_reply = server.rcon_exec "cvarlist"
      print "#{rcon_reply}\n"

      assert(
        rcon_reply.include?("CvarList ? for syntax"),
        "Did not receive complete cvarlist.")
    end
  end
  
  # This test tries to run the "status" command over RCON on a Source server
  def test_long_rcon_source_server
    assert_nothing_raised do
      server = SourceServer.new IPAddr.new("127.0.0.1")
      if server.rcon_auth "test"
        rcon_reply = server.rcon_exec "cvarlist"
        print "#{rcon_reply}\n"
      end
    end
  end

  # This test tries to run the "status" command over RCON on a GoldSrc server
  def test_short_rcon_goldsrc_server
    assert_nothing_raised do
      server = GoldSrcServer.new IPAddr.new("127.0.0.1")
      server.rcon_auth "test"
      rcon_reply = server.rcon_exec "version"
      print "#{rcon_reply}\n"

      assert(
        rcon_reply.include?("Protocol version") &&
        rcon_reply.include?("Exe version") &&
        rcon_reply.include?("Exe build"),
        "Did not receive correct version response.");
    end
  end
  
  # This test tries to run the "status" command over RCON on a Source server
  def test_short_rcon_source_server
    assert_nothing_raised do
      server = SourceServer.new IPAddr.new("127.0.0.1")
      if server.rcon_auth "test"
        rcon_reply = server.rcon_exec "version"
        print "#{rcon_reply}\n"
      end
    end
  end

end
